src/spec/datasource.test.ts

```diff
@@ -0,0 +1,1363 @@
+import * as _ from 'lodash';
+import q from 'q';
+
+jest.mock('app/core/utils/datemath');
+
+import CogniteDatasource, {
+  DataQueryRequest,
+  QueryTarget,
+  Tab,
+  VariableQueryData,
+} from '../datasource';
+import Utils from '../utils';
+
+function getDataqueryResponse(request: DataQueryRequest) {
+  const items = request.items;
+  const itemsArr = [];
+  const aggregation = Utils.getDatasourceValueString(request.aggregates);
+
+  for (const item of items) {
+    const datapoints = [1549336675000, 1549337275000, 1549337875000, 1549338475000].map(
+      (timestamp, i) => {
+        const obj = { timestamp };
+        obj[aggregation] = i;
+        return obj;
+      }
+    );
+    itemsArr.push({
+      datapoints,
+      name: item.name,
+    });
+  }
+  return getDataqueryResponseObject(itemsArr, request.aggregates ? aggregation : undefined);
+}
+
+function getDataqueryResponseObject(items, aggregates) {
+  const response = {
+    data: {
+      data: { items },
+    },
+    config: {
+      data: { aggregates },
+    },
+  };
+  return response;
+}
+function getTimeseriesResponse(items) {
+  const response = {
+    data: {
+      data: { items },
+    },
+  };
+  return response;
+}
+
+describe('CogniteDatasource', () => {
+  const ctx: any = {};
+  const instanceSettings = {
+    id: 1,
+    name: 'Cognite Test Data',
+    type: 'cognitedata-platform-datasource',
+    url: '/api/datasources/proxy/6',
```

Comment:
I don't know what this `/6` is, is it deterministic?

Reply:
Grafana uses that url to distinguish between the different datasources that are included, this just happened to be one that I copied from one of my test datasources. Doesn't actually do anything in these tests.